backend/eventloop/eventlooptypes/types.go

```diff
@@ -126,20 +128,78 @@ func GetWorkspaceIDFromNamespaceID(namespace corev1.Namespace) string {
 	return string(namespace.UID)
 }
 
-func GetK8sClientForGitOpsEngineInstance(gitopsEngineInstance *db.GitopsEngineInstance) (client.Client, error) {
+// GetK8sClientForGitOpsEngineInstance returns a client for accessing resources from a GitOpsEngine cluster based on the environment.
+// KCP environment: return a client that targets a workload cluster where the GitOpsEngine instance is deployed.
+// non-KCP environment: return a normal client that targets the same cluster as backend.
+func GetK8sClientForGitOpsEngineInstance(ctx context.Context, gitopsEngineInstance *db.GitopsEngineInstance) (client.Client, error) {
 
 	// TODO: GITOPSRVCE-73: When we support multiple Argo CD instances (and multiple instances on separate clusters), this logic should be updated.
 
+	if !sharedutil.IsRunningAgainstKCP() {
+		return GetK8sClientForServiceWorkspace()
+	}
+
 	config, err := sharedutil.GetRESTConfig()
 	if err != nil {
 		return nil, err
 	}
 
 	scheme := runtime.NewScheme()
+	err = corev1.AddToScheme(scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	k8sClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Could you add a couple short comments here briefing explaining why we are doing this, e.g. why we are retrieving this secret. Something like "this retrieves the information required to connect to the Argo CD instance, which is stored in the secret."